citeproc/source/bibtex/latex/__init__.py

```diff
@@ -86,16 +86,28 @@ def dispatch(tokens, macros, level=0):
                 warn("Unbalanced parenthesis in '{}'".format(tokens.string))
             break
         if next_token.type == OPEN_SCOPE:
-            yield handle_scope(tokens, macros, level)
+            try:
+                yield handle_scope(tokens, macros, level)
+            except StopIteration:
+                return
         elif next_token.type == CLOSE_SCOPE:
             raise ScopeClosing
         elif next_token.type == START_MACRO:
-            yield handle_macro(tokens, macros)
+            try:
+                yield handle_macro(tokens, macros)
```

Comment:
this could hide away the problems -- underlying `next()` invocations are within `assert` statements, e.g.


```python
def handle_macro(tokens, macros):
    assert next(tokens).type == START_MACRO
```

The fact that we are getting StopIteration there and masking it would pretty much disregard that assert and likely just hide away some underlying issue , result in "ignoring some data" etc.

Someone should just try to reproduce (`*`) and analyze why StopIteration and only if/where it is expected to get it -- there assert/except.

`*` Ideally such a fix here should start with adding a test which would depict this problem(s).

Reply:
Thanks for the hint! I invested some time to investigate and might isolated the root problem.